docs/topics/native/apple-privacy-manifest.md

```diff
@@ -51,5 +43,74 @@ The resulting file is a collection of dictionaries. For each accessed API type,
+of your Kotlin framework and a [separate plugin](#place-the-xcprivacy-file-in-your-kotlin-artifacts) to bundle
+`.xcprivacy` file with your Kotlin artifacts.
+
 If a new privacy manifest doesn't help satisfy App Store requirements or you cannot figure out how to go through the steps,
 contact us and share your case in [this YouTrack issue](https://youtrack.jetbrains.com/issue/KT-67603).
+
+## Find usages of required reason APIs
+
+Kotlin code in your app may access required reason APIs from libraries such as `platform.posix`.
+For example, if your Kotlin code or one of the dependencies accesses API such as `fstat`:
+
+```kotlin
+import platform.posix.fstat
+
+fun useRequiredReasonAPI() {
+    fstat(...)
+}
+```
+
+In some cases, it may be difficult to determine which dependency causes the required reason API usage.
+To help you find it, we've built a simple tool.
+
+To use it, run the following command in the directory where the Kotlin framework is declared in your project:
+
+```shell
+/usr/bin/python3 -c "$(curl -fsSL https://github.com/JetBrains/kotlin/raw/rrf_v0.0.1/libraries/tools/required-reason-finder/required_reason_finder.py)"
+```
+
+You may also download this script separately, inspect it, and run it using `python3`.
+
+## Place the `.xcprivacy` file in your Kotlin artifacts
+
+If you need to bundle the `PrivacyInfo.xcprivacy` file with your Kotlin artifacts, use the `apple-privacy-manifests` plugin:
+
+```kotlin
+plugins {
+    kotlin("multiplatform")
+    kotlin("apple-privacy-manifests") version "1.0.0"
+}
+
+kotlin {
+    privacyManifest {
+        embed(
+            privacyManifest = layout.projectDirectory.file("PrivacyInfo.xcprivacy").asFile,
+        )
+    }
+}
+```
+
+The plugin will copy the privacy manifest file to the [corresponding output location](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/adding_a_privacy_manifest_to_your_app_or_third-party_sdk?language=objc).
+
+## Known usages
+
+### Compose Multiplatform
+
+Using Compose Multiplatform may result in `fstat`, `stat` and `mach_absolute_time` usages in your binary.
+Even though these functions are not used for tracking or fingerprinting and are not sent from the device, Apple can still
+flag them as APIs with missing required reasons. 
+
+If you must specify a reason for `stat` and `fstat` usages, use `0A2A.1`. For `mach_absolute_time`, use `35F9.1`.
+
+For further updates on required reasons APIs used in Compose Multiplatform, follow [this issue](https://github.com/JetBrains/compose-multiplatform/issues/4738).
+
+### Kotlin/Native runtime in versions 1.9.10 or earlier
+
+The `mach_absolute_time` API is used in the `mimalloc` allocator in the Kotlin/Native runtime. This was the default
+allocator in Kotlin 1.9.10 and earlier versions.
+
+We recommend upgrading to Kotlin 1.9.20 or later versions or changing the [allocator settings](native-memory-manager.md#adjust-memory-consumption)
```

Comment:
As far as I understand, there isn't a range of settings to change, the user just needs to switch to the system allocator? We can probably say that right here, before the user follows the link and needs to figure out what exactly needs changing.